src/generate/ast/node.rs

```diff
@@ -89,3 +81,87 @@ pub enum Core {
     With { resource: Box<Core>, expr: Box<Core> },
     WithAs { resource: Box<Core>, alias: Box<Core>, expr: Box<Core> },
 }
+
+#[derive(Debug, PartialEq, Eq, Hash, Clone)]
+pub enum CoreOp {
+    Assign,
+    AddAssign,
+    SubAssign,
+    MulAssign,
+    DivAssign,
+    PowAssign,
+    BLShiftAssign,
+    BRShiftAssign,
+
+    Ge,
+    Geq,
+    Le,
+    Leq,
+    Eq,
+    Neq,
+    Add,
+    Sub,
+    Mul,
+    Div,
+    Pow,
+    Mod,
+    FDiv,
+}
+
+impl CoreOp {
+    pub fn maybe_from(lit: &str) -> Option<CoreOp> {
```

Comment:
check if `maybe_` prefix is idiomatic. Else, it is pretty obvious from the function returning `Option`.